src/extension.ts

```diff
@@ -203,7 +198,7 @@ export async function activate(context: vscode.ExtensionContext): Promise<void>
     const phanScriptPath = conf.get<string>('phanScriptPath') || defaultPhanScriptPath;
     // Support analyzing more than one project the simplest way (always start every server).
     const originalAnalyzedProjectDirectories = conf.get<string|string[]>('analyzedProjectDirectory');
-    const analyzedProjectDirectories = normalizeDirsToAnalyze(originalAnalyzedProjectDirectories);
+    let analyzedProjectDirectories = normalizeDirsToAnalyze(originalAnalyzedProjectDirectories);
```

Comment:
Another option I thought of would be to have the magic value `*` for phan.analyzedProjectDirectory mean that this extension should attempt to open any workspace with a valid phan config (in addition to hiding any showOpenSettingsPrompt calls).

Or just a boolean `phan.enableForAllProjects: true`, which is equivalent.

If that was enabled, then nothing would be popped up. If that was disabled, then phan would show the existing popup, but mention the new alternatives for making the popup not show up

That way, it'd be possible both to force Phan to start a server for a few directories, and to analyze the currently open directores (e.g. in case a workspace had multiple phan configs, or a user occasionally opened different projects)

Still, I'd really rather not enable this by default

- Projects may have their own plugins in `.phan/config.php` which aren't compatible with the version of phan bundled with vs code (or configured by the user in vs code settings for other projects)
- Projects may use Phan configuration settings that are extremely, extremely slow or memory intensive, or contain files that take a long time to parse/run plugins on

Reply:
If analyzedProjectDirectory was merged with findValidProjectDirectories(), the list of folders would need to be deduplicated, preferring the configured folder name.

https://nodejs.org/api/fs.html#fs_fs_realpathsync_path_options is one way, in case there are symlinks or different ways to represent the same folder

Reply:
I don't understand exactly what you mean here, I will try tomorrow, maybe there is a better solution for this to be worked out.

Reply:
> If analyzedProjectDirectory was merged with findValidProjectDirectories(), the list of folders would need to be deduplicated, preferring the configured folder name.

This I don't also understand, analyzedProjectDirectory is a setting and findValidProjectDirectories() is a function, if anything is passed into analyzedProjectDirectory, even a string that is not a path, then findValidProjectDirectories() will not run at all.